spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/orm/jpa/JpaBaseConfiguration.java

```diff
@@ -76,14 +79,21 @@
 
 	private ConfigurableListableBeanFactory beanFactory;
 
+	private Environment environment;
+
+	private RelaxedPropertyResolver datasourcePropertyResolver;
+
 	protected JpaBaseConfiguration(DataSource dataSource, JpaProperties properties,
 			ObjectProvider<JtaTransactionManager> jtaTransactionManager,
-			ObjectProvider<TransactionManagerCustomizers> transactionManagerCustomizers) {
+			ObjectProvider<TransactionManagerCustomizers> transactionManagerCustomizers,
+			Environment environment) {
 		this.dataSource = dataSource;
 		this.properties = properties;
 		this.jtaTransactionManager = jtaTransactionManager.getIfAvailable();
 		this.transactionManagerCustomizers = transactionManagerCustomizers
 				.getIfAvailable();
+		this.environment = environment;
+		this.datasourcePropertyResolver = new RelaxedPropertyResolver(this.environment, "spring.datasource.");
```

Comment:
How do you feel about retrieving the URL of the `DataSource` from the meta-data (like you did in that other PR).  This avoid injecting another collaborator and would work for things where the url isn't defined (that's typically the case for auto-configured embedded databases).